### PR TITLE
Add rbi definition for `Enumerator#+`

### PR DESCRIPTION
**Pull request by devosnw**
### Motivation

* Was added in Ruby 2.7.0
* https://ruby-doc.org/core-2.7.0/Enumerator.html#method-i-2B

### Test plan

I am not familiar with this repo at all. If I need to update some tests somewhere please let me know!

### CODE REVIEW

File: rbi/core/enumerator.rbi [hunk 1]
```diff
@@ -154,6 +154,18 @@ class Enumerator < Object
...
+  # Returns an enumerator object generated from this enumerator and a given 
+  # enumerable.
+  #
+  # ### Examples
+  #
+  # ```ruby
+  # e = (1..3).each + [4, 5]
+  # e.to_a #=> [1, 2, 3, 4, 5]
+  # ```
+  sig { params(enum: T::Enumerable[T.untyped]).returns(T::Enumerator::Chain[T.untyped]) }
```
Comment (jez) [inline rbi/core/enumerator.rbi:166]: 👋 Why does this not use generics to avoid depending on `T.untyped`?

**Review by jez — CHANGES_REQUESTED** [review]
^ question

File: rbi/core/enumerator.rbi (hunk 1 above)
Comment (devosnw) [reply rbi/core/enumerator.rbi:166]: Generics really throw me every time I have to work with them but from how I understand it the resulting chain would look something like

```suggestion
  sig { type_parameters(:T).params(enum: T::Enumerable[type_parameter(:T)]).returns(T::Enumerator::Chain[T.any(type_parameter(:T), self::Elem) }
```

What would be the correct way to write this?

Comment (jez) [reply rbi/core/enumerator.rbi:166]: yep, I believe so

Comment (jez) [reply rbi/core/enumerator.rbi:166]: You can omit the `self::`, just `Elem` is fine.

Comment (devosnw) [reply rbi/core/enumerator.rbi]: done! https://github.com/sorbet/sorbet/pull/8226/commits/5150cd820814775b0bbde29dec730a447cfbb2bd

**Comment by neilparikh** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_R0sUDpfeBncnJk
→ https://go/builds/bui_R0sUA6I6yn8kmy
→ https://go/builds/bui_R0sUQh9cW5UwnX

**Comment by jez** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_R2mgSnOK94bcvx
→ https://go/builds/bui_R2mgw0A5432PZy
→ https://go/builds/bui_R2mgmlWET2yN33

**Comment by jez** [comment]
Sorry for the delay. There was a problematic commit that had landed on Sorbet master that needed to be fixed, which made it look like your change introduced failures. We've resolved that, re-tested, and things look good.

Thanks for the ping and thanks for the contribution.

**Review by jez — APPROVED** [review]

**Pull request merged**